client/components/command-pallette/use-command-pallette.tsx

```diff
@@ -0,0 +1,92 @@
+import styled from '@emotion/styled';
+import { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+import { useSiteExcerptsQuery } from 'calypso/data/sites/use-site-excerpts-query';
+import { useAddNewSiteUrl } from 'calypso/lib/paths/use-add-new-site-url';
+import { generateCommandsArrayWpcom } from 'calypso/sites-dashboard/components/wpcom-smp-commands';
+
+const SiteImage = styled.img( {
+	height: 24,
+	width: 24,
+	minWidth: 24,
+	objectFit: 'cover',
+} );
+
+type CloseFunction = () => void;
+type OnClickSiteFunction = ( {
+	close,
+	site,
+}: {
+	close: CloseFunction;
+	site: SiteExcerptData;
+} ) => void;
+interface SiteFunctions {
+	onClick: ( { site, close }: { site: SiteExcerptData; close: CloseFunction } ) => void;
+	filter?: ( site: SiteExcerptData ) => boolean | undefined | null;
+}
+interface Command {
+	name: string;
+	label: string;
+	searchLabel: string;
+	callback: ( {
+		close,
+		setSearch,
+	}: {
+		close: CloseFunction;
+		setSearch: ( search: string ) => void;
+	} ) => void;
+	context?: string;
+	icon?: JSX.Element;
+	image?: JSX.Element;
+	siteFunctions?: SiteFunctions;
+	separator?: boolean;
+}
+interface UseSMPCommands {
```

Comment:
Can we rename the type to something like `UseCommandPaletteOptions` ?

Reply:
Done!